src/workerd/util/stream-utils.c++

```diff
@@ -171,11 +232,13 @@ kj::Own<kj::AsyncOutputStream> newNullOutputStream() {
   return kj::Own<kj::AsyncOutputStream>(&nullStream, kj::NullDisposer::instance);
 }
 
-kj::Own<kj::AsyncInputStream> newMemoryInputStream(kj::ArrayPtr<const kj::byte> data) {
-  return kj::heap<MemoryInputStream>(data);
+kj::Own<kj::AsyncInputStream> newMemoryInputStream(
+    kj::ArrayPtr<const kj::byte> data, kj::Maybe<kj::Own<void>> maybeBacking) {
+  return kj::heap<MemoryInputStream>(data, kj::mv(maybeBacking));
 }
 
-kj::Own<kj::AsyncInputStream> newMemoryInputStream(kj::StringPtr data) {
+kj::Own<kj::AsyncInputStream> newMemoryInputStream(
+    kj::StringPtr data, kj::Maybe<kj::Own<void>> maybeBacking) {
   return kj::heap<MemoryInputStream>(data.asBytes());
```

Comment:
haven't looked at the whole PR yet, but shouldn't maybeBacking be attached here too?